HeavyIonsAnalysis/JetAnalysis/src/JetMatcher.cc

```diff
@@ -0,0 +1,42 @@
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Common/interface/AssociationMap.h"
+#include "DataFormats/JetReco/interface/JetCollection.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+class JetMatcher : public edm::global::EDProducer<> {
+public:
+  typedef edm::AssociationMap<edm::OneToOne<reco::JetView, reco::JetView> > JetMatchMap;
+  explicit JetMatcher(const edm::ParameterSet& iConfig)
+      : sourceToken_(consumes<reco::JetView>(iConfig.getParameter<edm::InputTag>("source"))),
+        matchedToken_(consumes<reco::JetView>(iConfig.getParameter<edm::InputTag>("matched"))) {
+    produces<JetMatchMap>();
+  };
+  ~JetMatcher() override{};
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+private:
+  const edm::EDGetTokenT<reco::JetView> sourceToken_;
+  const edm::EDGetTokenT<reco::JetView> matchedToken_;
+};
+
+void JetMatcher::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  const auto& source = iEvent.getHandle(sourceToken_);
+  const auto& matched = iEvent.getHandle(matchedToken_);
+  auto matching = std::make_unique<JetMatchMap>(source, matched);
```

Comment:
I would add here a short comment summarizing the logic of the matching done below